jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java

```diff
@@ -14,112 +14,120 @@
-                "-nested.yaml",
-                ".properties",
-                "++suffix.yaml"}) {
-            Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + path);
-            assertThat(map).contains(
-                    entry("http.port", 80),
-                    entry("https.port", 443)
-            );
+    @Nested
+    @DisplayName("read config")
+    class ReadConfig {
+
+      @DisplayName("from valid file")
+      @ParameterizedTest(name = "''{0}'' file, config should be added")
+      @ValueSource(strings = { ".json", ".yaml", "-nested.yaml", ".properties", "++suffix.yaml" })
+      void readConfig(String path) {
+        Map<String, Integer> map = extractFromFile("vertx.config", "AbstractPortsExtractorTest" + path);
+        assertThat(map).contains(
+            entry("http.port", 80),
+            entry("https.port", 443));
+      }
+
+      @Test
+      @DisplayName("from invalid extension file, should throw exception")
+      void fromInvalidFileExtension_shouldThrowException() {
+        assertThatIllegalArgumentException()
+            .isThrownBy(() -> extractFromFile("vertx.config.test", "AbstractPortsExtractorTest" + ".bla"))
+            .withMessageContaining("Unknown file extension.");
+      }
+
+      @Test
+      @DisplayName("without file, config should be empty")
+      void withNoFile_shouldBeEmpty() {
+        Map<String, Integer> map = extractFromFile("vertx.config", null);
+        assertThat(map).isNotNull().isEmpty();
+      }
+
+      @Test
+      @DisplayName("from non-existing file, config should be empty")
+      void configFileDoesNotExist_shouldBeEmpty() {
+        final String nonExistingFile = "/bla/blub/lalala/config.yml";
+        System.setProperty("vertx.config.test", nonExistingFile);
+        try {
+          Map<String, Integer> map = extractFromFile("vertx.config.test", null);
+          assertThat(map).isNotNull().isEmpty();
+        } finally {
+          System.getProperties().remove("vertx.config.test");
         }
+      }
     }
 
-    @Test
-    public void testKeyPatterns() {
-        Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-keys.yml");
-
-        Object[] testData = {
-                "web.port", true,
-                "web_port", true,
-                "webPort", true,
-                "ssl.support", false,
-                "ports", false,
-                "ports.http", false,
-                "ports.https", false
-        };
-
-        for (int i = 0; i < testData.length; i +=2 ) {
-            assertEquals(testData[i+1], map.containsKey(testData[i]));
-        }
+    @ParameterizedTest(name = "''{0}'' should be ''{1}''")
+    @MethodSource("patterns")
+    void keyPatterns(String key, boolean expected) {
+      Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-keys.yml");
+      assertThat(map.containsKey(key)).isEqualTo(expected);
```

Comment:
I cannot understand why you're using a parameterized test to check map keys. 

Parameterized test are supposed to be used when there is a possibility of running same test with different input/outputs.

Reply:
I'm really sorry, it was due to some misunderstandings.
Fixed.